controllers/virtualmachine/virtualmachine_controller.go

```diff
@@ -165,6 +171,39 @@ func classBindingToVMMapperFn(ctx *context.ControllerManagerContext, c client.Cl
 	}
 }
 
+// classToVMMapperFn returns a mapper function that can be used to queue reconcile request
+// for the VirtualMachines in response to an event on the VirtualMachineClass resource when
+// WCP_Namespaced_Class_And_Windows_Support is enabled.
+func classToVMMapperFn(ctx *context.ControllerManagerContext, c client.Client) func(o client.Object) []reconcile.Request {
+	// For a given VirtualMachineClass, return reconcile requests
+	// for those VirtualMachines with corresponding VirtualMachinesClasses referenced
+	return func(o client.Object) []reconcile.Request {
+		class := o.(*vmopv1.VirtualMachineClass)
+		logger := ctx.Logger.WithValues("name", class.Name, "namespace", class.Namespace)
+
+		logger.V(4).Info("Reconciling all VMs referencing a VM class because of a VirtualMachineClass watch")
```

Comment:
If the mapper function is also called after resync period (every 10 minutes?) then on scale systems, it would result in log spew for _all_ VM classes even though they did not result in any reconciles.  What do you think about bumping the log level even further down?  

Separately, L202 is critical so I would vote for bumping that up.  It might help knowing the what VM class change resulted in a reconcile.

Reply:
Yeah, I bumped up to level `Info`, which is the default level.